kernel/src/tty/core_env.rs

```diff
@@ -35,6 +36,45 @@ pub fn init_core_env(env: &Rc<RefCell<Env>>) {
             env: env.clone(),
         },
     );
+    env.borrow_mut().set(
+        "read-string",
+        MalType::Builtin {
+            eval: read_string,
+            args: Box::new(MalType::List(vec![MalType::Symbol("a".to_string())])),
+            body: Box::new(MalType::Nil),
+            env: env.clone(),
+        },
+    );
+    env.borrow_mut().set(
+        "slurp",
+        MalType::Builtin {
+            eval: slurp,
+            args: Box::new(MalType::List(vec![MalType::Symbol("a".to_string())])),
+            body: Box::new(MalType::Nil),
+            env: env.clone(),
+        },
+    );
+    env.borrow_mut().set(
+        "ls",
+        MalType::Builtin {
+            eval: ls,
+            args: Box::new(MalType::List(vec![])),
+            body: Box::new(MalType::Nil),
+            env: env.clone(),
+        },
+    );
+    env.borrow_mut().set(
+        "write-to-file",
+        MalType::Builtin {
+            eval: write_to_file,
+            args: Box::new(MalType::List(vec![
+                MalType::Symbol("a".to_string()),
+                MalType::Symbol("b".to_string()),
```

Comment:
Are the arguments names 'a' and 'b' going to appear to the user in any way? 
```suggestion
                MalType::Symbol("file".to_string()),
                MalType::Symbol("content".to_string()),
```

Reply:
Nope these are hidden from the user, there is no risk of collision with a user symbol as no user code is run in those builtin functions. Giving them better names is a great idea, thanks!